crates/wit-parser/src/ast.rs

```diff
@@ -1,24 +1,329 @@
+                            let interface = match interface_map.get(&*id.name) {
+                                Some(interface) => interface.clone(),
+                                None => {
+                                    return Err(Error {
+                                        span: id.span,
+                                        msg: format!("{} not defined in document", id.name),
+                                    }.into());
+                                },
+                            };
+                            if world.imports.insert(name.to_string(), interface).is_some() {
+                                return Err(Error {
+                                    span: span,
+                                    msg: format!("duplicate import name {}", name),
+                                }
+                                .into());
+                            }
+                        },
+                    }
+
+                }
+                // resolve the exports
+                for Export { docs, name, kind } in wit.exports() {
+                    let (name, span) = (&name.name, name.span);
+
+                    match kind {
+                        ExternKind::Interface(items) => {
+                            let mut resolver = resolve::Resolver::default();
+                            let interface = resolver.resolve("", &items, &HashMap::new())?;
+
+                            if world.exports.insert(name.to_string(), interface).is_some() {
+                                return Err(Error {
+                                    span: span,
+                                    msg: format!("duplicate export name {}", name),
+                                }
+                                .into());
+                            }
+                        },
+                        // ExternKind::Func(function) => {
+                        //     todo!("world-export-resolve-value");
+                        // },
+                        ExternKind::Id(id) => {
+                            let interface = match interface_map.get(&*id.name) {
+                                Some(interface) => interface.clone(),
+                                None => {
+                                    return Err(Error {
+                                        span: id.span,
+                                        msg: format!("{} not defined in document", id.name),
+                                    }.into());
+                                },
+                            };
+                            if world.exports.insert(name.to_string(), interface).is_some() {
+                                return Err(Error {
+                                    span: span,
+                                    msg: format!("duplicate export name {}", name),
+                                }
+                                .into());
+                            }
+                        },
```

Comment:
This looks pretty similar to the import case so could this be refactored to a helper method?